### PR TITLE
Add Turbolinks

### PR DESCRIPTION
**Pull request by bruncun**
[Turbolinks](https://github.com/turbolinks/turbolinks) makes navigating Chalk even faster. For only 33 KB, you get an SPA that renders additional page views ~2x faster. There's a small learning curve to Turbolinks, but I find the benefits far outweigh the costs. This PR will manually add Turbolinks and restructure the existing JS for compatibility.

Check it out and let me know what you think. I'm also planning to strip jQuery, load all JS async, and  slightly enhance the page transition (by adding a fade out on link clicks). If any of this sounds interesting, I can send additional pull requests.

### CODE REVIEW

**Comment by nielsenramon** [comment]
@bruncun I like the idea of using turbolinks for sure. I'm just a bit hesitant of ditching jQuery as this is a template for everyone and people tend to know jQuery very well ;-)

I would love to see everything ported over to turbolinks with the async js for sure! Can you continue on this PR? Also try importing turbolinks with bower eventually?

Afterwards I can add documentation on how to add jQuery if they wish to.

**Comment by bruncun** [comment]
@nielsenramon You can see all my performance changes [live on my blog](http://brunocunha.net). Includes:

- Turbolinks
- jQuery removal (replaced fluidbox with [baguetteBox.js](https://feimosi.github.io/baguetteBox.js/)
- Inline CSS (cumbersome manual process)
- Webfonts removed
- Icon set replaced with necessary inline SVG icons

Loading the JS async not only added no obvious added performance, but it also caused some funky/tricky behavior from both Turbolinks and Google Analytics, so I just opted to load it normally in the header. Like dropping jQuery, async would be more trouble than its worth for most.

All in all, I got the initial page load down to 2 requests and like ~20 kb, gzipped. Really pleased with the results.

I hadn't considered jQuery's convenience. Did you know that adding Turbolinks would require that developers stop installing behavior with jQuery's ready event, and instead opt for listening to "turbo links:load"? It's a nuance, but more than enough to cause headaches for anyone expecting the typical jQuery workflow.

🤔

**Comment by nielsenramon** [comment]
@bruncun Great to see your version life! Looking good.
The main issue I am having with your solution is that the JavaScript is loaded in the head. Which PageSpeed punishes a lot. If that would be fixed it would be a no-brainer for me to get turbolinks in.

- I really like the fluidbox transition here (mainly a visual choice) so not really into replacing it with yours.
- Inline CSS is something I though about to, maybe this is something for the build process. Will look into it.
- I am also planning on replacing the webfonts with a custom icon set for Chalk.

Is it really such a hassle to load all the JS Async? I guess that would fix the PageSpeed punishment.

**Comment by migueldemoura** [comment]
@bruncun, I've made a few of these changes to my blog as well - local webfonts, no jQuery, and svg icons. I don't understand, however, why you would inline css given that you lose the caching benefit on multiple page loads.

All in all I welcome the changes, especially since turbolinks degrades nicely with no-js.

**Comment by bruncun** [comment]
@migueldemoura There are tradeoffs to be sure. In addition to 1 less request, by inlining CSS we avoid blocking the critical rendering path. It's worth noting that, on subsequent visits, Turbolinks merges the new page's content into the existing window, eliminating the need to reload stylesheets. Also, our inline CSS still gets cached inside our HTML assets. I don't expect mine to change often, and my gzipped total payload is ~25kb so I'm personally okay with this caching strategy. YMMV.

Its awesome to see some excitement about Turbolinks! I should be wrapping this branch up this weekend.

**Comment by migueldemoura** [comment]
> Turbolinks merges the new page's content into the existing window

Didn't dive deep into Turbolink's spec, but if that's the case then we should be fine.

**Comment by migueldemoura** [comment]
@bruncun, can you add `rel="noopener noreferrer"` to all `</a >` with `target=_blank`?

Issue POC: lcamtuf.coredump.cx/switch/

**Comment by bruncun** [comment]
@nielsenramon Branch updates!

- Use Bower to install Turbolinks. Had to create a "turbolinks5" package because the original is taken and author seems to have gone AWOL.
- As requested, return Javascript include to end of body tag to avoid PageSpeed penalties. From a Turbolinks perspective, this should be fine. I believe including Turbolinks this way will cause Turbolinks to attempt load on every visit, but everything should work just fine.
- I encountered some issues with the social share links. The onClick handlers were breaking with Turbolinks. Rather than preventing Turbolinks on these links, I figured this could be a good refactor.

File: _layouts/post.html [hunk 1]
```diff
@@ -38,17 +38,17 @@
...
-              <span class="icon icon-social-googleplus"></span>
+            <a href="https://plus.google.com/share?url={{ page.url | absolute_url }}" title="Share on Google+" target="_blank">
+              <svg enable-background="new 0 0 128 128" version="1.1" viewBox="0 0 128 128" width="15px" xml:space="preserve" xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink"><g id="_x35__stroke"><g id="Google_Plus"><rect clip-rule="evenodd" fill="none" fill-rule="evenodd" height="128" width="128"/><path clip-rule="evenodd" d="M40.654,55.935v16.13    c0,0,15.619-0.021,21.979-0.021C59.189,82.5,53.834,88.194,40.654,88.194c-13.338,0-23.748-10.832-23.748-24.194    s10.41-24.194,23.748-24.194c7.052,0,11.607,2.483,15.784,5.944c3.344-3.35,3.065-3.828,11.573-11.877    c-7.222-6.586-16.822-10.6-27.357-10.6C18.201,23.273,0,41.507,0,64c0,22.493,18.201,40.727,40.654,40.727    c33.561,0,41.763-29.275,39.044-48.792H40.654z M113.912,56.742V42.628h-10.063v14.113H89.358v10.081h14.491v14.517h10.063V66.823    H128V56.742H113.912z" fill-rule="evenodd" id="Google_Plus_1_"/></g></g></svg>
             </a>
           </div>
-
+
```
Comment (jdiago) [inline _layouts/post.html]: Careful with the trailing whitespace. 😈

Comment (bruncun) [inline _layouts/post.html]: https://github.com/nielsenramon/chalk/pull/40/commits/4fbef36c46e93996dcaaeeafd23b905bca457bd6

**Comment by bruncun** [comment]
@migueldemoura Nice catch. https://github.com/nielsenramon/chalk/pull/40/commits/4097166ae65dc31cefb6827dd3471126047ccb81

**Comment by nielsenramon** [comment]
@bruncun Nice work! I like the changes. Gonna test out all the basic features before merging this one (pagespeed, google analytics, ...). 

Would there be any way to open a window when clicking the social sharing buttons instead of a new tab? I made them that way so they opened the correct size of the share block :-)

**Comment by bruncun** [comment]
@nielsenramon Done! Whatcha think?

**Comment by nielsenramon** [comment]
@bruncun Sorry for taking so long. Big thanks!

**Comment by bruncun** [comment]
No problem, happy to give back!

**Comment by migueldemoura** [comment]
@bruncun, are you planning on submitting further pull requests to remove webfonts and replace ionicons with svg icons?

**Comment by bruncun** [comment]
Not at this time, sorry.

**Pull request merged**